lib/reducers/user_reducer.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:redux/redux.dart';
+
+import '../actions/actions.dart';
+import '../models/models.dart';
+
+final userReducer = TypedReducer<User, dynamic>(_updateUserReducer);
+
+User _updateUserReducer(User user, dynamic action) {
+  if (action is UpdateUser) {
+    return action.payload;
+  }
+  if (action is UpdateFullName) {
+    return user.copyWith(fullName: action.payload);
+  }
+  if (action is UpdateSummary) {
+    return user.copyWith(summary: action.payload);
+  }
+  if (action is AddSchool) {
+    return user.copyWith(
+      schools: [
+        ...?user.schools,
+        action.payload
+      ]
+    );
+  }
+  if (action is UpdateSchool) {
+    var schools = user.schools.map((school) {
+      if (school.id == action.payload.id) {
+        return action.payload;
+      }
+      return school;
+    }).toList();
```

Comment:
This kinda bothers me since we're iterating through the whole list just to update one item, but I haven't yet found a more efficient way to do it here on Flutter! I guess we don't have to worry about this right now but it's definitely something to think about later on (IF we're going to have larger lists for schools, jobs, and skills)...

Reply:
There's a couple of `where` list methods you can try. There's some details on them [here](https://bezkoder.com/dart-list/), specifically, under the `Filter items for a List in Dart/Flutter` heading.

Reply:
Oooo, I'll check these out! Thank youuu